bsip-0084.md

```diff
@@ -0,0 +1,195 @@
+    * must exist
+    * must not be present in the user's voting slate on `authority`
+    * if `authority.candidates` is not empty then it must be contained therein
+    * if `authority.candidate_holders` is present then it must own tokens of the given type
+* For all entries in `votes_to_remove`:
+    * must be present in the user's voting slate on `authority`
+* If both `votes_to_add` and `votes_to_remove` are empty then `number` must be different than the `voter`'s previous choice for `number`, or the `voter`'s proxy setting must change.
+* If `proxy` is present then
+    * `authority.proxy_allowed` must be `true`.
+    * `proxy` account must exist and must have a voting slate for the `authority`.
+    * `number` must equal 0 and both `votes_to_add` and `votes_to_remove` must be empty.
+
+**Evaluation:**
+
+* If `proxy` is not present but `voter` had set a proxy on this `authority` before
+    * Subtract the `voter`'s token balance from the old proxy's proxy token count and adjust vote tally accordingly.
+    * Apply `votes_to_add` as described below.
+* If `proxy` is present and `voter` had not set a proxy before
+    * Remove all accounts from the user's own voting slate and adjust vote tally accordingly.
+    * Set proxy in voting slate.
+    * Add `voter`'s balance to `proxy`'s proxy token count and adjust vote tally accordingly.
+* If `proxy` is present and `voter` had set a (different) proxy before
+    * Remove old proxy and adjust vote tally accordingly (see above).
+    * Set new proxy and adjust vote tally accordingly (see above).
+* If `proxy` is not set
+    * `votes_to_add`, `votes_to_remove` and `number` are applied to the user's voting slate.
+    * Let `voting_balance` be the sum of the `voter`'s token balance of the asset plus his proxy token balance.
+    * The voting delta is the `voting_balance` for each vote to be added and for the new `number`, the negative `voting_balance` for each vote to be removed and the previous `number`.
+    * The voting delta in votes is applied to the vote tally of `authority`.
+    * The resulting differences in the election outcome (if any) are reflected in the respective `authority` object.
+
+**Note:** the intent is to have vote tallying work in the same way it is currently performed when voting with BTS. The same goes for determining the number of accounts that make up the authority, unless it is fixed.
+
+
+## 5. `adjust_balance`
+
+The chain logic for adjusting account balances is modified as follows:
+
+* For each voting slate of the balance's owner with the same asset type as is being updated:
+    * If the voting slate points to a proxy, adjust the proxy's proxy token count according to the balance delta.
+    * Apply the balance delta to the vote tally according to the user's (or proxy's if set) voting slate.
+    * Reflect the resulting differences to the election outcome in the respective `authority` object.
+
+
+## 6. `account_update_operation`
```

Comment:
How about account_create_operation?

Reply:
I wanted to keep it simple.

Reply:
Won't support? I'm fine with it. Perhaps add a note.

Reply:
Done.